### DISCUSSION TITLE
Livewire is not found

### DISCUSSION THREAD

**Discussion by ArtyomBist** [filamentphp/filament#9816] [Help]
when I do use not different domain name than Localhost for local dev I have the issue "Livewire is not found" 
![image](https://github.com/filamentphp/filament/assets/10978770/27645a98-77a9-4818-916b-1c79386127c0)
Steps to reproduce
try to use https://ospanel.io/ for example, create a default Laravel project, install Livewire, and filament, set up a domain name (see the scr), and try to log in
![image](https://github.com/filamentphp/filament/assets/10978770/c81f4b96-a43d-486d-8360-9d629c31eace)

**Comment by kombiahrk** [comment]
Today i faced the same issue when Uploading the filament Project to live server

This link will help you

**https://stackoverflow.com/questions/73053717/login-issue-in-laravel-9-x-with-filament**

**Comment by sivarajd** [comment]
Livewire JS file routes are generated dynamically.
![image](https://github.com/user-attachments/assets/40e71393-0d06-4392-b9a7-0b520171a8f2)

Do no serve JS files directly from the webserver. Let Laravel handle them.

**Comment by enessvg** [comment]
Run Commands:
`php artisan livewire:publish --config `
`php artisan livewire:publish --assets `

**Reply by siubie** [reply]
this is works but i dont knowh what causing it

**Reply by rizalabulfata** [reply]
setuju Pak, saya menemukan saat deploy filament v4 dan v3 test on frankenphp

**Comment by rohititaliaya** [comment]
I have the same issue 
this file is not found /livewire/livewire.js on production only , on my local environment its working.

any one found the real cause ?

**Comment by aurimasbutkus** [comment]
I encountered this issue as well and it was caused by this: https://livewire.laravel.com/docs/4.x/upgrading#livewire-asset-and-endpoint-url-changes.
I was optimizing routes with `php artisan optimize` during deployment, and livewire's routes now depend on APP_KEY, so the routes it cached were different from what they actually were once the app is deployed. You can solve this by adding `--except=routes` flag to the optimize command during deployment or cache the routes after the app is deployed.

**Reply by stuartcusackie** [reply]
Same problem. I have three or four filament 4 projects so it's strange that only one of them stopped working today. I think this may be down to Laravel Forge's latest deploy script for zero-downtime which includes the `php artisan optimize` command by default, unlike previous versions.

`--except=routes`  does what it says, you can try it locally first. Hopefully that solves it but I will report back if not.
